### DISCUSSION TITLE
Map ctrl+e/n/s inside init.lua?

### DISCUSSION THREAD

**Discussion by Nikola-Milovic** [vscode-neovim/vscode-neovim#1947] [Q&A]
I tried mapping `<c-e` in neovim but that just doesnt work, unable to get it to trigger, so I am guessing these keys arent sent to neovim but I have no idea why.

So I decided to try and set a keybinding via composite keys

```json
   "vscode-neovim.compositeKeys": {
       "<c-e>": {
           "command": "vscode-harpoon.editEditors"
       },
```

Which gives me the below error

`Invalid composite key: <c-e>. Composite key must be exactly 2 ASCII characters long.`.

Preferably I'd somehow do these inside neovim `init.lua`

```lua
-- Harpoon
map("n", "<c-e>", function()
  vim.notify("Harpoon c-e")
  vscode.call("vscode-harpoon.editEditors")
end, { remap = true })
map("n", "<c-n>", function()
  vim.notify("Harpoon c-n")
  vscode.call("vscode-harpoon.gotoEditor3")
end, { remap = true })
map("n", "<c-s>", function()
  vim.notify("Harpoon c-s")
  vscode.call("vscode-harpoon.gotoEditor4")
end, { remap = true })
```

I deleted everything non-vscode neovim related from keybindings

![screenshot of keybindings](https://github.com/vscode-neovim/vscode-neovim/assets/50072027/bddbf203-d157-4604-bc17-bc41027344af)

**Comment by xiyaowong** [comment]
Add this to the "keybindings.json":
```json
[
  {
    "key": "ctrl+e",
    "command": "vscode-neovim.send",
    "args": "<C-e>",
    "when": "editorTextFocus && neovim.init"
  }
]
```

https://github.com/vscode-neovim/vscode-neovim#add-keybindings


"composite keys" are only for needs such as:
Use jj or jk to leave insert mode
```vim
imap jj <esc>
imap jk <esc>
```

**Reply by Nikola-Milovic** [reply]
Still not triggering, not sure what I am I missing, seems simple enough

```json
[
    {
        "key": "ctrl+e",
        "command": "vscode-neovim.send",
        "args": "<C-e>",
        "when": "editorTextFocus && neovim.init"
    }
]
```

And in neovim 

```lua
map("n", "<c-e>", function()
  vim.notify("Harpoon c-e")
  vscode.call("vscode-harpoon.editEditors")
end, { remap = true })
```

**Reply by xiyaowong** [reply]
Need a recording, including your configuration and how you use 
 the shortcuts.

**Reply by Nikola-Milovic** [reply]
@xiyaowong Not sure how to record to showcase this, but here's a minimal reproduction repo.

`keybindings.json`
```json
[
    {
        "key": "ctrl+e",
        "command": "vscode-neovim.send",
        "args": "<C-e>",
        "when": "editorTextFocus && neovim.init"
    }
]
```


And the `init.lua`

```lua
local vscode = require("vscode-neovim")
if vim.g.vscode then
  vim.notify = vscode.notify
end

vim.g.mapleader = " "

local function map(mode, lhs, rhs, opts)
  opts = opts or {}
  opts.silent = opts.silent ~= false
  if opts.remap and not vim.g.vscode then
    opts.remap = nil
  end
  vim.keymap.set(mode, lhs, rhs, opts)
end

vim.notify("Loaded nvim config")

map("n", "<C-e>", function()
  vim.notify("Harpoon c-e")
end, {})
```

You can try pressing `ctrl+e`, and no notification will pop up

**Reply by theol0403** [reply]
Did you update ctrlKeysForNormalMode

**Reply by Nikola-Milovic** [reply]
@theol0403 What exactly needs to be updated?

```json
"vscode-neovim.ctrlKeysForNormalMode": [
  "a",
  "b",
  "c",
  "d",
  "e",
  "f",
  "h",
  "i",
  "j",
  "k",
  "l",
  "m",
  "o",
  "r",
  "t",
  "u",
  "v",
  "w",
  "x",
  "y",
  "z",
  "/",
  "]",
  "right",
  "left",
  "up",
  "down",
  "backspace",
  "delete"
]
```

`e` is listed so I guess that is okay?

**Comment by xiyaowong** [comment]
> Not sure how to record to showcase this, but here's a minimal reproduction repo.

I mainly want to know if you use this keybinding when focusing on the text editor.

**Reply by Nikola-Milovic** [reply]
@xiyaowong yeah I am inside the text editor in normal mode. `ctrl+h` works for example, just not ctrl+e and ctrl+n/s
